WIP (#14)
MichaelCurrin/url-manager: docs/browser_onetab_extraction.md

The approach below parses the `storage.json` and gets the value of 'state' field inside it.

1. Open Firefox.
2. Go to the _about:profiles_ page. This will show you your Firefox users.
3. Choose the profile you want, look at the paths and copy the username from one. e.g. `abcd1234.default`
4. Follow the commands below to enter the browser and username and save the output. An example is shown below. Set your username as the second argument.
    ```bash
    $ cd url_manager
    $ source venv/bin/activate
    # TODO: Make this part of script and config
    # See also debug
    # NB. This no longer works due to the Firefox OneTab migration.
    $ ./extract_onetab_storage.py Firefox abcd1234.default > var/lib/raw/onetab_firefox_abc_personal.json
    ```
5. Go back to step 3 and repeat for other profiles as desired.

Resources:

> - [Profiles - Where Firefox stores your bookmarks, passwords and other user data](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data)

Author: ```suggestion

- [Profiles - Where Firefox stores your bookmarks, passwords and other user data](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data)
```

The approach below parses the `storage.json` and gets the value of 'state' field inside it.

1. Open Firefox.
2. Go to the _about:profiles_ page. This will show you your Firefox users.
3. Choose the profile you want, look at the paths and copy the username from one. e.g. `abcd1234.default`
4. Follow the commands below to enter the browser and username and save the output. An example is shown below. Set your username as the second argument.
    ```bash
    $ cd url_manager
    $ source venv/bin/activate

>     # TODO: Make this part of script and config

Author: ```suggestion
```

>     # See also debug

Author: ```suggestion
```

>     # NB. This no longer works due to the Firefox OneTab migration.

Author: ```suggestion
    # FIXME. Note this no longer works due to the Firefox OneTab migration.
```

        > path/to/repo/url_manager/var/lib/raw/chrome_onetab_mycompany_personal.json
    $ # You can view the file if you want.
    $ view path/to/repo/url_manager/var/lib/raw/chrome_onetab__mycompany_personal.json
    ```

>     ```json

Author: ```suggestion
    ```json5
```
